libraries/stdlib/src/kotlin/text/Strings.kt

```diff
@@ -1162,3 +1162,97 @@ public fun CharSequence.lineSequence(): Sequence<String> = splitToSequence("\r\n
  * * Splits this char sequence to a list of lines delimited by any of the following character sequences: CRLF, LF or CR.
  */
 public fun CharSequence.lines(): List<String> = lineSequence().toList()
+
+/**
+ * For each i in [0, length), this function computes
+ * the length of the longest suffix of a substring of pattern from 0 to i
+ * that is also a prefix of the pattern itself.
+ */
+private fun computePrefixFunction(pattern: CharSequence): IntArray {
```

Comment:
I've fixed the infinite loop, the prefix function algorithm missed a subtraction that wasn't caught by a test yet, so I've also added a test case for that.

I've never seen -1 in the table before, so KMP as described in the Wikipedia article seems to be a slightly different variation of KMP. My implementation is based on the pseudocode we were given in a lecture on Fundamental Algorithms at the University of Paderborn and it was also proven to be working correct by the lecturer, so this should be fine.